### ISSUE TITLE
Fail to install on Python 3.10

### ISSUE THREAD

**Issue by haobibo** [fafhrd91/aiohttp#6898]
### Describe the bug

aiohttp install on Python 3.10.6 reports bug when compiling _http_parser

### To Reproduce

Install on Python 3.10.6 on linux with: pip install aiohttp==4.0.0a1

### Expected behavior

aiohttp can be successfully installed on Python 3.10.6

### Logs/tracebacks
Refer to: https://github.com/QPod/docker-images/runs/8052701739?check_suite_focus=true#step:3:1199

```python-traceback
Fail to install on Python3.10 with the following compiling error:

      aiohttp/_http_parser.c: In function ‘__pyx_tp_dealloc_7aiohttp_12_http_parser_HttpParser’:
      aiohttp/_http_parser.c:16227:5: error: lvalue required as increment operand
      16227 |     ++Py_REFCNT(o);
            |     ^~
      aiohttp/_http_parser.c:16229:5: error: lvalue required as decrement operand
      16229 |     --Py_REFCNT(o);
```
```



### aiohttp Version

```console
4.0.0a1
```

### OS

Linux

### Related component

Server, Client

### Code of Conduct

- [X] I agree to follow the aio-libs Code of Conduct

**Comment by davfsa** [comment]
aiohttp 4.0.0a1 was released back in 2019. The most stable version is (and might be for a while) 3.8.1. I recommend you switch to it, as it will most likely have all the functionality you need, newest bugfixes and work out of the box. Also, see #6799

![image](https://user-images.githubusercontent.com/29100934/187424645-1fc8ef59-5b4b-4c0b-815b-2edeb05bc000.png)

**Comment by akira-kurogane** [comment]
To share my experience:
I encountered this "lvalue required as increment operand" error on the `++Py_REFCNT(o)` statement in aiohttp/_http_parser.c whilst doing `[sudo] python3 setup.py install` within the source checkout of karapace 3.3.2 on Ubuntu 22 today. (gcc 11.2.0, python 3.10.4). The current dependency in karapace is for aiohttp==3.8.1.

However when I did a global install of aiohttp first using `[sudo] pip3 install aiohttp==3.8.1` there was no error.

After that I could finish the install of karapace, using the same setup.py as I had done in the first place, with no further build/install issues appearing.

I haven't yet had any good idea why. Only a vague guess that maybe it's a compiler -xxx-no-warning-... flag that suppressed newer GCC's behaviour.
